### PR TITLE
chore: add documentation for the deprcated link macro

### PR DESCRIPTION
**Pull request by schalkneethling**
This adds documentation on replacing any remaining instances of the {{link}} macro.

Part of https://github.com/mdn/mdn/issues/241

### CODE REVIEW

File: files/en-us/mdn/writing_guidelines/page_structures/macros/deprecated_macros/index.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+{{MDNSidebar}}
+
+The following is a list of [KumaScript macros](https://github.com/mdn/yari/tree/main/kumascript/macros) that has been deprecated and should no longer be used. This document also provides guidance on how to replace these macros should you encounter them in the content.
+
+## The `Link` macro
+
+The Link macro has [been deprecated](https://github.com/mdn/yari/pull/6865) and
+should no longer be used. Any occurances that still exist in actively maintained
```
Comment (bsmth) [inline files/en-us/mdn/writing_guidelines/page_structures/macros/deprecated_macros/index.md:17]: typo

```suggestion
should no longer be used. Any occurrences that still exist in actively-maintained
```

File: files/en-us/mdn/writing_guidelines/page_structures/macros/deprecated_macros/index.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+content should be replaced with a plain Markdown style link.
```
Comment (bsmth) [inline files/en-us/mdn/writing_guidelines/page_structures/macros/deprecated_macros/index.md:18]:
```suggestion
content should be replaced with a plain Markdown-style link (`[Example link](https://example.com)`).
```

File: files/en-us/mdn/writing_guidelines/page_structures/macros/deprecated_macros/index.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+---
+title: How to replace deprecated Kumascript macros
```
Comment (bsmth) [inline files/en-us/mdn/writing_guidelines/page_structures/macros/deprecated_macros/index.md:2]:
```suggestion
title: How to replace deprecated KumaScript macros
```

**Review by bsmth — APPROVED** [review]
Thanks @schalkneethling , I added some notes that I think would help

File: files/en-us/mdn/writing_guidelines/page_structures/macros/deprecated_macros/index.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  - writing-guide
+---
+
+{{MDNSidebar}}
+
+The following is a list of [KumaScript macros](https://github.com/mdn/yari/tree/main/kumascript/macros) that has been deprecated and should no longer be used. This document also provides guidance on how to replace these macros should you encounter them in the content.
+
+## The `Link` macro
+
+The Link macro has [been deprecated](https://github.com/mdn/yari/pull/6865) and
```
Comment (bsmth) [inline files/en-us/mdn/writing_guidelines/page_structures/macros/deprecated_macros/index.md:16]:
```suggestion
The `Link` macro [is now deprecated](https://github.com/mdn/yari/pull/6865) and
```

**Review by Josh-Cena — COMMENTED** [review]
General question: who is this page intended for? Most contributors would not even know its existence, and (I believe) writers from old KS would not actively use most linking macros either (apart from xrefs), so concerns about introducing new `{{Link}}` calls seem mostly immaterial—a deprecation flaw is enough to tell people to switch to plain Markdown links. OTOH, I believe macros are deprecated only after they have been successfully removed from the content repo? Is it intended only for translations? What do we do after they have been removed from translations as well (which should take a short time)?

**Comment by schalkneethling** [comment]
> General question: who is this page intended for? Most contributors would not even know its existence, and (I believe) writers from old KS would not actively use most linking macros either (apart from xrefs), so concerns about introducing new `{{Link}}` calls seem mostly immaterial—a deprecation flaw is enough to tell people to switch to plain Markdown links. OTOH, I believe macros are deprecated only after they have been successfully removed from the content repo? Is it intended only for translations? What do we do after they have been removed from translations as well (which should take a short time)?

This is the first of many to help with macro deprecation work. Some folks will run into situations like this, but the only feedback they get is, "This macro has been deprecated and should no longer be used". This particular one is pretty straightforward to replace, but that is not the case for all macros.

Once this lands, the `deprecated` call inside the Macro will be updated to point to this document. So instead of the default message, it would point you towards guidance on replacing the macro with the alternative we want them to use.

**Comment by Josh-Cena** [comment]
Got it. I guess my point is more like: under which circumstance will such a message be shown and the writer be asked to fix it?

- Do we anticipate someone to "accidentally" write these macros? (Most contributors don't know much about macros)
- Do we want writers to fix the existing macros on pages that they touch and hence come up in the review companion? (Their changes usually are orthogonal to the occurrences of these macros)
- Will the macros appear at all? For example, I don't think `{{Link}}` exists in content anymore (which is why we set out to deprecate it). Will this workflow change in the future—lie, deprecate first, publish a migration guide, and then ask individual contributors to gradually fix them, rather than as a one-time chore?

**Comment by schalkneethling** [comment]
> Got it. I guess my point is more like: under which circumstance will such a message be shown and the writer be asked to fix it?
> 
> * Do we anticipate someone to "accidentally" write these macros? (Most contributors don't know much about macros)
> * Do we want writers to fix the existing macros on pages that they touch and hence come up in the review companion? (Their changes usually are orthogonal to the occurrences of these macros)
> * Will the macros appear at all? For example, I don't think `{{Link}}` exists in content anymore (which is why we set out to deprecate it). Will this workflow change in the future—lie, deprecate first, publish a migration guide, and then ask individual contributors to gradually fix them, rather than as a one-time chore?

All good questions :)

> * Do we anticipate someone to "accidentally" write these macros? (Most contributors don't know much about macros)

I do not think that is too much of a concern.

> * Do we want writers to fix the existing macros on pages that they touch and hence come up in the review companion? (Their changes usually are orthogonal to the occurrences of these macros)

So, when marked as deprecated, it will throw a flaw error, and the information about the macro will be output to the page. I guess it will also show up in the PR companion. You might know about this already, but for the benefit of others, here is the code: https://github.com/mdn/yari/blob/cce98ca0e11e1fcfd808ee60ffd643ea942b479c/kumascript/src/api/mdn.ts#L127

> Will the macros appear at all? For example, I don't think `{{Link}}` exists in content anymore (which is why we set out to deprecate it).

There are still ~3-4 instances of Link that are being addressed. There are many others, however, that is still on many pages that are also actively being replaced. The thing is, we could perhaps have had more people fix these if we provided clear guidance on how to replace a macro. I know I have run into some and had to do some digging to understand how it would be replaced.

That, then, is one of the main aims here. Provide a link to clear documentation on how to replace the macro right in the context where a contributor encounters a deprecated macro.

**Comment by MrBrain295** [comment]
I’m in the process of replacing the last few occurrences so this probably won’t be needed and will just be deleted soon after it’s merged so though I’m all for the depreciated macro documentation I don’t think this is a good one to start with

**Comment by schalkneethling** [comment]
> I’m in the process of replacing the last few occurrences so this probably won’t be needed and will just be deleted soon after it’s merged so though I’m all for the depreciated macro documentation I don’t think this is a good one to start with
> 
> Edit: I’ve made a pr to remove the last few link occurrences

Thank you for the feedback and for removing the last few instances. I reckon I am going to go ahead and merge this just to get things rolling, so to speak, and get started on the next one.

PS: I will most likely delete this one with the next PR :)

**Pull request merged**